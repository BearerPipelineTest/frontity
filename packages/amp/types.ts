import { Package, Action, MergePackages } from "frontity/types";
import Html2React from "@frontity/html2react/types";
import Source from "@frontity/source/types";

/**
 * The types of the AMP Frontity package.
 */
interface AMP extends Package {
  /**
   * The name of this package.
   */
  name: "@frontity/amp";

  /**
   * Actions exposed by this package.
   */
  actions: {
    /**
     * Amp namespace.
     */
    amp: {
      /**
       * An internal action that bootstraps the initialization.
       *
       * @remarks
       * This action is not meant to be run by the user, but by the Frontity
       * framework.
       */
      init: Action<Packages>;
    };
  };
}

/**
 * Packages used internally by the AMP package.
 */
export type Packages = MergePackages<AMP, Source, Html2React>;

export default AMP;
