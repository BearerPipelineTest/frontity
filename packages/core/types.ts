import { TransformOptions } from "@babel/core";
import { Configuration } from "webpack";

export type Mode = "development" | "production";
export type Target = "module" | "es5" | "server";

export interface BabelConfigs {
  module: TransformOptions;
  es5: TransformOptions;
  server: TransformOptions;
}

export interface WebpackConfigs {
  module: Configuration;
  es5: Configuration;
  server: Configuration;
}

export interface FrontityConfig {
  outDir: string;
}

export interface Config {
  babel: BabelConfigs;
  webpack: WebpackConfigs;
  frontity: FrontityConfig;
}

export interface EntryPoints {
  name: string;
  path: string;
}

export type FrontityTags = {
  script?: string;
  link?: string;
  style?: string;
};

export type HeadTags = {
  bodyAttributes: string;
  htmlAttributes: string;
  head: string[];
};

export type Template = ({
  html,
  scripts,
  head,
  htmlAttributes,
  bodyAttributes,
}: {
  html: string;
  scripts: [];
  head: [];
  htmlAttributes?: string;
  bodyAttributes?: string;
}) => string;
