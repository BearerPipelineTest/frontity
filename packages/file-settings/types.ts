/**
 * Abstract base type used to make the {@link Multi} and {@link Normalized} types.
 */
type Base = {
  /**
   * Field to distinguish between different sites.
   */
  match?: string[];

  /**
   * The mode of this site.
   *
   * Used to choose to render regular HTML (default) or AMP-compatible HTML.
   */
  mode?: "default" | "amp";

  /**
   * Global state.
   */
  state?: Record<string, unknown>;
};

/**
 * The base type of a Frontity package.
 */
export type Package = {
  /**
   * Package name.
   */
  name: string;

  /**
   * Whether the package should be active or not.
   */
  active?: boolean;

  /**
   * Package state.
   */
  state?: Record<string, unknown>;
};

/**
 * Package that has been normalized.
 */
type NormalizedPackage = Required<Package>;

/**
 * Multi-site package.
 */
type Multi = Base & {
  /**
   * Site name.
   */
  name: string;
};

/**
 * Represents normalized site settings.
 */
type Normalized<T> = Base & {
  /**
   * Mode of the site.
   */
  mode: "default" | "amp";

  /**
   * Site state.
   */
  state: Record<string, unknown>; // Default: {}

  /**
   * Packages of the site.
   */
  packages: T[];
};

/**
 * Normalized site settings.
 */
export type NormalizedSettings<T = NormalizedPackage> = Normalized<T> & Multi;

/**
 * A Frontity site.
 */
export type Site = {
  /**
   * Site name.
   */
  name: string;

  /**
   * The mode of this site.
   */
  mode: "default" | "amp";

  /**
   * Site packages.
   */
  packages: string[];
};
