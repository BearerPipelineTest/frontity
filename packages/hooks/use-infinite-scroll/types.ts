import useInView from "../use-in-view";
import { MergePackages } from "frontity/types";
import Source from "@frontity/source/types";
import Router from "@frontity/router/types";

/**
 * The intersection options passed down to `useInView`.
 *
 * They come originally from the
 * [`react-intersection-observer`](https://github.com/thebuilder/react-intersection-observer)
 * package.
 */
export type IntersectionOptions = Parameters<typeof useInView>[0];

/**
 * Options of the {@link useInfiniteScroll} hook.
 */
export type UseInfiniteScrollOptions = {
  /**
   * The current link that should be used to start the infinite scroll.
   */
  currentLink: string;

  /**
   * The next link that should be fetched and loaded once the user scrolls
   * down.
   */
  nextLink?: string;

  /**
   * The options for the internal {@link useInView} hook used for the
   * `actions.source.fetch`.
   */
  fetchInViewOptions?: IntersectionOptions;

  /**
   * The options for the internal {@link useInView} hook used for the
   * `actions.router.set`.
   */
  routeInViewOptions?: IntersectionOptions;
};

/**
 * Options of the {@link useInfiniteScroll} hook.
 */
export type UseInfiniteScrollOutput =
  | {
      /**
       * Boolean indicating if the Intersection Observer is supported or not by
       * the browser.
       */
      supported: false;

      /**
       * The ref that should be attached to the element used to trigger
       * `actions.router.set`.
       */
      routeRef?: undefined;

      /**
       * The ref that should be attached to the element used to trigger
       * `actions.source.fetch`.
       */
      fetchRef?: undefined;

      /**
       * Boolean that indicates when the element used to trigger
       * `actions.router.set` is in the screen.
       */
      routeInView?: undefined;

      /**
       * Boolean that indicates when the element used to trigger
       * `actions.source.fetch` is in the screen.
       */
      fetchInView?: undefined;
    }
  | {
      /**
       * Boolean indicating if the Intersection Observer is supported or not by
       * the browser.
       */
      supported: true;

      /**
       * The ref that should be attached to the element used to trigger
       * `actions.router.set`.
       */
      routeRef: (node?: Element) => void;

      /**
       * The ref that should be attached to the element used to trigger
       * `actions.source.fetch`.
       */
      fetchRef: (node?: Element) => void;

      /**
       * Boolean that indicates when the element used to trigger
       * `actions.router.set` is in the screen.
       */
      routeInView: boolean;

      /**
       * Boolean that indicates when the element used to trigger
       * `actions.source.fetch` is in the screen.
       */
      fetchInView: boolean;
    };

/**
 * Properties added by the {@link useInfiniteScroll} to the Router
 * state property.
 */
export type InfiniteScrollRouterState = {
  /**
   * The infinite scroll map of props.
   */
  infiniteScroll: {
    /**
     * Array of visited links.
     */
    links: string[];

    /**
     * Number of elements that will be automatically fetched.
     */
    limit: number;

    /**
     * List of pages by link. TODO: check this type.
     */
    pages: string[];

    /**
     * Archive (used by `usePostTypeInfiniteScroll`).
     */
    archive?: string;
  };

  /**
   * Any other prop.
   */
  [key: string]: unknown;
};

/**
 * The type of those packages the {@link useInfiniteScroll} hook depends on,
 * merged together.
 */
export type Packages = MergePackages<Source, Router>;
