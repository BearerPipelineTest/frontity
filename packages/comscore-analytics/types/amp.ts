import React from "react";
import { Package } from "frontity/types";

/**
 * Interface for AMP support.
 */
export default interface ComscoreAnalyticsAmp extends Package {
  /**
   * Root elements exported by this package.
   */
  roots: {
    /**
     * The comScore namespace.
     */
    comscoreAnalytics: React.ElementType;
  };

  /**
   * The state exposed by this package.
   */
  state: {
    /**
     * Comscore Analytics namespace.
     */
    comscoreAnalytics: {
      /**
       * Single tracking ID associated to a Comscore client.
       */
      trackingId?: string;

      /**
       * Array of tracking IDs associated to a Comscore client.
       */
      trackingIds?: string[];
    };
  };
}
