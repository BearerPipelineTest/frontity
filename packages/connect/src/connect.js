import { createContext, useContext, Component } from "react";
import { view } from "@frontity/react-easy-state";
import { warn } from "@frontity/error";

/**
 * A React context that stores the store of Frontity Connect.
 */
const context = createContext({});

/**
 * The Provider of the store context, to be used in the root of the React
 * application.
 */
export const Provider = context.Provider;

/**
 * A flag to know when a connected component is being rendered.
 */
let isConnected = false;

/**
 * The default options of connect.
 */
const defaultOptions = {
  injectProps: true,
};

const isStateless = (Comp) =>
  !(Comp.prototype && Comp.prototype.isReactComponent);

/**
 * Connects a React component with the Frontity connect store.
 *
 * @param Comp - The React component to be connected.
 * @param options - TO BE LINKED WHEN WE SWITCH TO TYPESCRIPT.
 *
 * @returns The same Component, but with the store included in the props.
 */
export const connect = (Comp, options) => {
  options = options ? { ...defaultOptions, ...options } : defaultOptions;

  if (isStateless(Comp)) {
    return view((props) => {
      const { state, actions, libraries } = useContext(context);
      isConnected = true;
      const rendered = Comp({
        ...props,
        ...(options.injectProps ? { state, actions, libraries } : {}),
      });
      isConnected = false;
      return rendered;
    });
  } else {
    if (options.injectProps) {
      /**
       * A wrapper for the Component that extracts the store from the context
       * and passes it to the Component.
       */
      class ConnectedComponent extends Component {
        /**
         * The render method of this React Class.
         *
         * @returns The same Component but with the store injected as props.
         */
        render() {
          const { state, actions, libraries } = this.context;
          const props = { ...this.props, state, actions, libraries };
          return <Comp {...props} />;
        }
      }
      ConnectedComponent.contextType = context;
      return view(ConnectedComponent);
    }
    return view(Comp);
  }
};

export const useConnect = () => {
  if (!isConnected)
    warn(
      "Warning: useConnect() is being used in a non connected component, " +
        "therefore the component won't update on state changes. " +
        "Please wrap your component with connect().\n"
    );

  return useContext(context);
};
