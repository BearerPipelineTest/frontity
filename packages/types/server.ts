import { Middleware, Next, ParameterizedContext } from "koa";
import { ResolveState, ResolveActions } from "./utils";
import Package from "./package";

/**
 * Frontity context for Koa middleware.
 */
export interface FrontityContext<Packages extends Package>
  extends ParameterizedContext {
  /**
   * Frontity namespace for custom properties.
   */
  frontity: any;

  /**
   * Packages state.
   */
  state: ResolveState<Packages["state"]>;

  /**
   * Packages actions.
   */
  actions: ResolveActions<Packages["actions"]>;

  /**
   * Packages libraries.
   */
  libraries: Packages["libraries"];

  /**
   * Packages server middleware.
   */
  server: Packages["server"];

  /**
   * Extended Koa context.
   */
  ctx: FrontityContext<Packages>;

  /**
   * Koa Next function.
   */
  next: Next;
}

/**
 * Types for server middleware in Frontity packages.
 */
export type Server<Packages extends Package> = Middleware<
  ResolveState<Packages["state"]>,
  FrontityContext<Packages>
>;

/**
 * Types for async server middleware in Frontity packages.
 */
export type AsyncServer<Packages extends Package> = Server<Packages>;
