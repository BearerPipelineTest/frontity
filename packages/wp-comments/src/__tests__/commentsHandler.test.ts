import { createStore, InitializedStore } from "@frontity/connect";
import wpSource from "@frontity/wp-source/src";
import clone from "clone-deep";
import { mergeDeepRight } from "ramda";
import wpComments from "../";
import { Packages } from "../../types";
import { mockResponse, mockComment } from "./mocks/helpers";

type Api = Packages["libraries"]["source"]["api"];

let store: InitializedStore<Packages>;
let api: jest.Mocked<Api>;

beforeEach(() => {
  const packages: any = mergeDeepRight(clone(wpSource()), clone(wpComments));
  store = createStore<Packages>(packages);
  store.state.source.api = "https://test.frontity.org/wp-json";
  store.actions.source.init();
  api = store.libraries.source.api as jest.Mocked<Api>;
  api.get = jest.fn();
});

describe("commentsHandler", () => {
  test("should populate comments correctly", async () => {
    api.get.mockResolvedValueOnce(
      mockResponse([
        mockComment({ post: 60, id: 2 }),
        mockComment({ post: 60, id: 1 }),
      ])
    );

    await store.actions.source.fetch("@comments/60");

    expect(api.get).toHaveBeenCalledTimes(1);
    expect(api.get.mock.calls[0]).toMatchSnapshot();
    expect(store.state.source.comment).toMatchSnapshot();
    expect(store.state.source.get("@comments/60")).toMatchSnapshot();
  });

  test("should work fine if there are no comments", async () => {
    api.get.mockResolvedValueOnce(mockResponse([]));

    await store.actions.source.fetch("@comments/60");

    expect(api.get).toHaveBeenCalledTimes(1);
    expect(api.get.mock.calls[0]).toMatchSnapshot();
    expect(store.state.source.comment).toMatchSnapshot();
    expect(store.state.source.get("@comments/60")).toMatchSnapshot();
  });

  test("should request more pages if there are more than 100 comments", async () => {
    // We use these headers to pretend that the responses contain many more
    // comments than they actually have.
    const responseProps = {
      headers: {
        "X-WP-TotalPages": "3",
        "X-WP-Total": "201",
      },
    };

    api.get
      .mockResolvedValueOnce(
        mockResponse([mockComment({ post: 60, id: 201 })], responseProps)
      )
      .mockResolvedValueOnce(
        mockResponse([mockComment({ post: 60, id: 101 })], responseProps)
      )
      .mockResolvedValueOnce(
        mockResponse([mockComment({ post: 60, id: 1 })], responseProps)
      );

    await store.actions.source.fetch("@comments/60");

    expect(api.get).toHaveBeenCalledTimes(3);
    expect(api.get.mock.calls[0]).toMatchSnapshot();
    expect(store.state.source.comment).toMatchSnapshot();
    expect(store.state.source.get("@comments/60")).toMatchSnapshot();
  });

  test("should handle replies correctly", async () => {
    api.get.mockResolvedValueOnce(
      mockResponse([
        mockComment({ post: 60, id: 3, parent: 2 }),
        mockComment({ post: 60, id: 2 }),
        mockComment({ post: 60, id: 1 }),
      ])
    );

    await store.actions.source.fetch("@comments/60");

    expect(api.get).toHaveBeenCalledTimes(1);
    expect(api.get.mock.calls[0]).toMatchSnapshot();
    expect(store.state.source.comment).toMatchSnapshot();
    expect(store.state.source.get("@comments/60")).toMatchSnapshot();
  });

  test("should handle replies correctly when more than one page", async () => {
    // We use these headers to pretend that the responses contain many more
    // comments than they actually have.
    const responseProps = {
      headers: {
        "X-WP-TotalPages": "2",
        "X-WP-Total": "103",
      },
    };

    api.get
      .mockResolvedValueOnce(
        mockResponse(
          [
            mockComment({ post: 60, id: 3, parent: 2 }),
            mockComment({ post: 60, id: 2 }),
            mockComment({ post: 60, id: 1 }),
          ],
          responseProps
        )
      )
      .mockResolvedValueOnce(
        mockResponse(
          [
            mockComment({ post: 60, id: 6 }),
            mockComment({ post: 60, id: 5, parent: 4 }),
            mockComment({ post: 60, id: 4, parent: 3 }),
          ],
          responseProps
        )
      );

    await store.actions.source.fetch("@comments/60");

    expect(api.get).toHaveBeenCalledTimes(2);
    expect(api.get.mock.calls[0]).toMatchSnapshot();
    expect(store.state.source.comment).toMatchSnapshot();
    expect(store.state.source.get("@comments/60")).toMatchSnapshot();
  });
});
