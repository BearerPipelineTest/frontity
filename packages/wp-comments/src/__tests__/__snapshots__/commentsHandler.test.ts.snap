// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commentsHandler should handle replies correctly 1`] = `
Array [
  Object {
    "endpoint": "comments",
    "params": Object {
      "_embed": false,
      "page": 1,
      "per_page": 100,
      "post": 60,
    },
  },
]
`;

exports[`commentsHandler should handle replies correctly 2`] = `
Object {
  "1": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 1</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 1,
    "link": "/2020/post-60/comment-page-1/#comment-1",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "2": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 2</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 2,
    "link": "/2020/post-60/comment-page-1/#comment-2",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "3": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 3</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 3,
    "link": "/2020/post-60/comment-page-1/#comment-3",
    "meta": Object {},
    "parent": 2,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
}
`;

exports[`commentsHandler should handle replies correctly 3`] = `
Object {
  "isComments": true,
  "isFetching": false,
  "isReady": true,
  "items": Array [
    Object {
      "children": Array [
        Object {
          "id": 3,
          "type": "comment",
        },
      ],
      "id": 2,
      "type": "comment",
    },
    Object {
      "id": 1,
      "type": "comment",
    },
  ],
  "link": "@comments/60/",
  "page": 1,
  "postId": 60,
  "query": Object {},
  "route": "@comments/60/",
  "total": 3,
  "totalPages": 1,
  "type": "comments",
}
`;

exports[`commentsHandler should handle replies correctly when more than one page 1`] = `
Array [
  Object {
    "endpoint": "comments",
    "params": Object {
      "_embed": false,
      "page": 1,
      "per_page": 100,
      "post": 60,
    },
  },
]
`;

exports[`commentsHandler should handle replies correctly when more than one page 2`] = `
Object {
  "1": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 1</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 1,
    "link": "/2020/post-60/comment-page-1/#comment-1",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "2": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 2</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 2,
    "link": "/2020/post-60/comment-page-1/#comment-2",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "3": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 3</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 3,
    "link": "/2020/post-60/comment-page-1/#comment-3",
    "meta": Object {},
    "parent": 2,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "4": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 4</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 4,
    "link": "/2020/post-60/comment-page-1/#comment-4",
    "meta": Object {},
    "parent": 3,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "5": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 5</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 5,
    "link": "/2020/post-60/comment-page-1/#comment-5",
    "meta": Object {},
    "parent": 4,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "6": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 6</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 6,
    "link": "/2020/post-60/comment-page-1/#comment-6",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
}
`;

exports[`commentsHandler should handle replies correctly when more than one page 3`] = `
Object {
  "isComments": true,
  "isFetching": false,
  "isReady": true,
  "items": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "id": 5,
                  "type": "comment",
                },
              ],
              "id": 4,
              "type": "comment",
            },
          ],
          "id": 3,
          "type": "comment",
        },
      ],
      "id": 2,
      "type": "comment",
    },
    Object {
      "id": 1,
      "type": "comment",
    },
    Object {
      "id": 6,
      "type": "comment",
    },
  ],
  "link": "@comments/60/",
  "page": 1,
  "postId": 60,
  "query": Object {},
  "route": "@comments/60/",
  "total": 103,
  "totalPages": 2,
  "type": "comments",
}
`;

exports[`commentsHandler should populate comments correctly 1`] = `
Array [
  Object {
    "endpoint": "comments",
    "params": Object {
      "_embed": false,
      "page": 1,
      "per_page": 100,
      "post": 60,
    },
  },
]
`;

exports[`commentsHandler should populate comments correctly 2`] = `
Object {
  "1": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 1</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 1,
    "link": "/2020/post-60/comment-page-1/#comment-1",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "2": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 2</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 2,
    "link": "/2020/post-60/comment-page-1/#comment-2",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
}
`;

exports[`commentsHandler should populate comments correctly 3`] = `
Object {
  "isComments": true,
  "isFetching": false,
  "isReady": true,
  "items": Array [
    Object {
      "id": 2,
      "type": "comment",
    },
    Object {
      "id": 1,
      "type": "comment",
    },
  ],
  "link": "@comments/60/",
  "page": 1,
  "postId": 60,
  "query": Object {},
  "route": "@comments/60/",
  "total": 2,
  "totalPages": 1,
  "type": "comments",
}
`;

exports[`commentsHandler should request more pages if there are more than 100 comments 1`] = `
Array [
  Object {
    "endpoint": "comments",
    "params": Object {
      "_embed": false,
      "page": 1,
      "per_page": 100,
      "post": 60,
    },
  },
]
`;

exports[`commentsHandler should request more pages if there are more than 100 comments 2`] = `
Object {
  "1": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 1</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 1,
    "link": "/2020/post-60/comment-page-1/#comment-1",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "101": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 101</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 101,
    "link": "/2020/post-60/comment-page-1/#comment-101",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
  "201": Object {
    "author": 1,
    "author_avatar_urls": Object {},
    "author_name": "Frontibotito",
    "author_url": "http://test.frontity.org",
    "content": Object {
      "rendered": "<p>Test Comment 201</p>",
    },
    "date": "2020-07-20T12:00:00",
    "date_gmt": "2020-07-20T10:00:00",
    "id": 201,
    "link": "/2020/post-60/comment-page-1/#comment-201",
    "meta": Object {},
    "parent": 0,
    "post": 60,
    "status": "approved",
    "type": "comment",
  },
}
`;

exports[`commentsHandler should request more pages if there are more than 100 comments 3`] = `
Object {
  "isComments": true,
  "isFetching": false,
  "isReady": true,
  "items": Array [
    Object {
      "id": 201,
      "type": "comment",
    },
    Object {
      "id": 101,
      "type": "comment",
    },
    Object {
      "id": 1,
      "type": "comment",
    },
  ],
  "link": "@comments/60/",
  "page": 1,
  "postId": 60,
  "query": Object {},
  "route": "@comments/60/",
  "total": 201,
  "totalPages": 3,
  "type": "comments",
}
`;

exports[`commentsHandler should work fine if there are no comments 1`] = `
Array [
  Object {
    "endpoint": "comments",
    "params": Object {
      "_embed": false,
      "page": 1,
      "per_page": 100,
      "post": 60,
    },
  },
]
`;

exports[`commentsHandler should work fine if there are no comments 2`] = `Object {}`;

exports[`commentsHandler should work fine if there are no comments 3`] = `
Object {
  "isComments": true,
  "isFetching": false,
  "isReady": true,
  "items": Array [],
  "link": "@comments/60/",
  "page": 1,
  "postId": 60,
  "query": Object {},
  "route": "@comments/60/",
  "total": 0,
  "totalPages": 1,
  "type": "comments",
}
`;
